### PR TITLE
Make a session time scrolling sticky

### LINKED ISSUE #493: Sticky time scroll in session list
**Idea Description**
Can we scroll sticky?
I would like to scroll in session list not in timetable.

**Reference images and links**
image

https://www.figma.com/file/NcSMs6dMsD88d4wOY0g3rK/DroidKaigi-2022-Conference-App?node-id=0%3A1

### PR DESCRIPTION
**Pull request by ryu38**
## Issue
- close #493

## Overview (Required)
- Update a scrolling behavior in session list to stick a session time at the top of the list until replaced by a different one.

## Links
- Nothing

## Screenshot

Before

[session_time_scroll.webm](https://user-images.githubusercontent.com/58480791/191659665-eb92a32a-1ab7-454b-84ff-35069a98a17c.webm)

After

[sticky_scroll.webm](https://user-images.githubusercontent.com/58480791/191659106-b7374f29-7efc-4ede-a69a-e0b4505548e2.webm)

### CODE REVIEW

File: feature/sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/SessionList.kt [hunk 1]
```diff
@@ -1,31 +1,99 @@
...
-        itemsIndexed(timetable) { _, item ->
-            key(item.second.timetableItem.id.value) {
-                content(item)
+        val visibleItemsInfo = remember {
+            derivedStateOf {
+                sessionsListListState.layoutInfo.visibleItemsInfo
+            }
+        }
+        var currentDurationTime: DurationTime? = null
+        visibleItemsInfo.value.forEachIndexed { visibleItemIndex, visibleItemInfo ->
```
Comment (takahirom) [inline feature/sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/SessionList.kt:45]: Thanks for your great effort!
Probably this cause performance problem.
I think it is tough. But I came up with two approaches to solve this problem.

1. Use draw behind 
In this way, you can skip the composition and layout phase

```kotlin
        LazyColumn(
            modifier = Modifier.fillMaxSize().drawBehind {
                     ...
            }
```

3. Pass sessionsListListState and consume it in the function

This way, you can't skip the recompose, but you can narrow the recompose.

```kotlin
fun TimeLane(sessionsListListState, timetable) {
        val visibleItemsInfo = remember {
            derivedStateOf {
                sessionsListListState.layoutInfo.visibleItemsInfo
            }
        }
}

Comment (ryu38) [reply feature/sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/SessionList.kt]: I'm sorry for my late response.
Your advice helped me for fixing the code, thank you!
I adopted the second one in my code (I tried solving this by using drawBehind but I can't ...).

Comment (takahirom) [reply feature/sessions/src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/SessionList.kt]: Thanks!

**Comment by takahirom** [comment]
Can you run ./gradlew spotlessKotlinApply?

**Comment by takahirom** [comment]
```
Step 'ktlint' found problem in 'src/main/java/io/github/droidkaigi/confsched2022/feature/sessions/TimeLane.kt':
Error on line: 36, column: 1
rule: max-line-length
Exceeded max line length (100)
java.lang.AssertionError: Error on line: 36, column: 1
```

**Comment by takahirom** [comment]
Please run ./gradlew composeLint and fix the issues?🙇

**Review by takahirom — APPROVED** [review]
Looks good! Thank you for your contribution!

**Pull request merged**